src/Core/Config/Version.cpp

```diff
@@ -0,0 +1,30 @@
+#include <Config/Exceptions.hpp>
+#include <Config/Version.hpp>
+
+#include <Utils/StringUtils.hpp>
+
+#include <string>
+
+namespace obe::Config
+{
+    Version::Version(const std::string& str)
+    {
+        std::vector<std::string> parts = Utils::String::split(str, ".");
+        if (parts.size() != 3)
+        {
+            throw Exceptions::InvalidVersionNumber(
+                "Wrong number of components for version string. Expecting version "
+                "string of the form v?MAJOR.MINOR.PATCH",
+                EXC_INFO);
+        }
+        // Skip first letter from version string ( 'v' )
+        if (!Utils::String::isStringNumeric(parts[0]) && parts[0].size() >= 2)
```

Comment:
So technically I could instanciate version with : 
```cpp
Version a("x0.1.1");
```
?

Reply:
Yep, do you think we should be stricter?

Reply:
I think so !